lib/queryBuilder/QueryBuilder.js

```diff
@@ -229,15 +229,7 @@ class QueryBuilder extends QueryBuilderBase {
   }
 
   eager(exp, filters) {
-    this._eagerExpression = exp || null;
-
-    if (isString(this._eagerExpression)) {
-      this._eagerExpression = parseRelationExpression(this._modelClass, this._eagerExpression);
-    }
-
-    if (filters) {
-      this._eagerExpression.filters = filters;
-    }
+    this._eagerExpression = parseRelationExpression(this._modelClass, exp, filters);
```

Comment:
@koskimas the `isString()` check here isn't really necessary because `RelationExpression.parse()` already does the same.